### PR TITLE
Extract spark from airflow container

### PR DESCRIPTION
**Pull request by krisgeus**

### CODE REVIEW

File: docker/airflow-python/delete_all_airflow_connections.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env python
+"""
+Removing all current airflow connections
+"""
+
+from airflow import settings
+from airflow.models import Connection
+
+session = settings.Session()
```
Comment (Fokko) [inline docker/airflow-python/delete_all_airflow_connections.py]: I'm a big fan of the create session `contextmanager`. I'm not against using the plain session here, it is just a short lived script, but I like to give good examples for people to use the `create_session()`
```suggestion
from airflow.utils.db import create_session
with create_session() as session:
session = settings.Session()
    session.query(Connection).delete()
```

File: docker/airflow-python/delete_all_airflow_connections.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+session.query(Connection).delete()
+session.commit()
+session.close()
+exit()
```
Comment (Fokko) [inline docker/airflow-python/delete_all_airflow_connections.py]: The exit is redundant.

File: docker/airflow-python/delete_all_airflow_connections.py (hunk 1 above)
Comment (krisgeus) [reply docker/airflow-python/delete_all_airflow_connections.py]: Done. In both scripts

Comment (Fokko) [reply docker/airflow-python/delete_all_airflow_connections.py]: Awesome! 👍

**Comment by bjgbeelen** [comment]
Looking great though!

File: envs/external-airflow-db/docker-compose.yml [hunk 3]
```diff
@@ -2,11 +2,12 @@ version: '3'
...
-    build:
-      context: ../../docker/airflow
-      dockerfile: Dockerfile
-      args:
-        - PYTHON_VERSION=${PYTHON_VERSION}
+#    build:
+#      context: ../../docker/airflow
+#      dockerfile: Dockerfile
+#      args:
+#        - PYTHON_VERSION=${PYTHON_VERSION}
```
[the remark this answers is not in the document]
Comment (krisgeus) [reply envs/external-airflow-db/docker-compose.yml]: Done

File: envs/airflow-s3-logging/.whirl.env [hunk 4]
```diff
@@ -4,3 +4,8 @@ S3_LOG_BUCKET=airflow-s3-logs
 AWS_SERVER=s3server
 AWS_PORT=4563
 PORT_WEB_UI=8080
+
+# Airflow variables
+AIRFLOW__CORE__EXPOSE_CONFIG=True
```
[the remark this answers is not in the document]
Comment (krisgeus) [reply envs/airflow-s3-logging/.whirl.env]: Yes it would make sense.

[the remark this answers is not in the document]
Comment (krisgeus) [reply envs/airflow-s3-logging/.whirl.env]: So tested it but it doesn't work. The top level .whirl.env is loaded in the whirl script but not inside the docker-compose files in each env.

Leaving it for now

File: docker/airflow-python/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+ENV AIRFLOW_APT_DEPS="libsasl2-dev sasl2-bin libsasl2-2 libsasl2-dev libsasl2-modules freetds-bin build-essential default-libmysqlclient-dev apt-utils curl rsync netcat locales"
+
+ENV AIRFLOW_VERSION=1.10.3
+ENV AIRFLOW_HOME /usr/local/airflow
+ENV AIRFLOW_GPL_UNIDECODE=yes
+ENV SLUGIFY_USES_TEXT_UNIDECODE=yes
+ENV WHIRL_SETUP_FOLDER=/etc/airflow/whirl.setup.d
+
+RUN mkdir -p /usr/share/man/man1 \
+  && update-ca-certificates -f \
```
[the remark this answers is not in the document]
Comment (krisgeus) [reply docker/airflow-python/Dockerfile]: some copy paste errors from dockerfile to script left RUN commands. Removed them. Can you check again? @bjgbeelen

File: docker/airflow-python/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,90 @@
+FROM python:3.6-slim as airflow-base
+
+ENV AIRFLOW_BUILD_DEPS="freetds-dev python-dev libkrb5-dev libssl-dev libffi-dev libpq-dev git"
+ENV AIRFLOW_APT_DEPS="libsasl2-dev sasl2-bin libsasl2-2 libsasl2-dev libsasl2-modules freetds-bin build-essential default-libmysqlclient-dev apt-utils curl rsync netcat locales"
+
+ENV AIRFLOW_VERSION=1.10.3
```
[the remark this answers is not in the document]
Comment (krisgeus) [reply docker/airflow-python/Dockerfile]: Leaving that for separate change

[the remark this answers is not in the document]
Comment (bjgbeelen) [reply docker/airflow-python/Dockerfile]: 👍

File: envs/airflow-s3-logging/.whirl.env (hunk 4 above)
[the remark this answers is not in the document]
Comment (bjgbeelen) [reply envs/airflow-s3-logging/.whirl.env]: Ok makes sense 9although a pity :-))

**Comment by bjgbeelen** [comment]
Maybe create a tag of the current master and then delete the `old airflow container code` in this PR as well?

File: docker/airflow-python/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,90 @@
+FROM python:3.6-slim as airflow-base
```
[the remark this answers is not in the document]
Comment (krisgeus) [reply docker/airflow-python/Dockerfile]: Will check.

[the remark this answers is not in the document]
Comment (krisgeus) [reply docker/airflow-python/Dockerfile]: Works. Pushed final changes

**Pull request merged**